### PR TITLE
- changed 'media not found' message into a dialog letting the user se…

### PR DESCRIPTION
**Pull request by wfeldt**
…lect

  the URL to retry (bsc #923918)

### CODE REVIEW

File: linuxrc.c [hunk 1]
```diff
@@ -1148,53 +1147,75 @@ void lxrc_init()
...
-    util_disp_init();
+    // ok, we failed to find a suitable repo
+    // do something about it
 
-    i = 0;
-    j = 1;
+    // If the file '/nextmedia' exists, get the message to show from there
+    // and retry the default repo settings if the user is ready.
+    // Otherwise, present the user the list of repos we have tried so far and
+    // let her choose / edit one and try again.
```
Comment (jreidinger) [inline linuxrc.c]: :+1: for explaining rationale

File: linuxrc.c [hunk 2]
```diff
@@ -1148,53 +1147,75 @@ void lxrc_init()
...
 
-      strprintf(&buf, "%s\n\nRetry?", s ?: "Please make sure your installation medium is available.");
-      do {
-        j = dia_yesno(buf, YES) == YES ? 1 : 0;
-        if(j) {
-          slist_t *sl;
+    do {
+      config.manual = 0;
+
+      if(config.cd1texts) {
```
Comment (jreidinger) [inline linuxrc.c]: work-flow in this if statement is a bit confusing as it actually do not use look from do ( always break at the end ).
So maybe it deserve own method and work-flow like:

```
config.manual = 0;
if(config.cd1texts) {
  ....
} else {
  do {
     ...
  }
}
```
of course unless I overlook that else branch can set `config.cd1texts`,  but from current code it doesn't look like.

File: linuxrc.c [hunk 3]
```diff
@@ -1567,3 +1588,71 @@ void lxrc_readd_parts()
 }
 
 
+/*
+ * Offer the user a list of URLs to choose from based in the current install
+ * URL and the default repo setting.
+ *
+ * In addidtion she can also edit the URL.
```
Comment (jreidinger) [inline linuxrc.c]: typo `addidtion`

File: linuxrc.c [hunk 4]
```diff
@@ -1567,3 +1588,71 @@ void lxrc_readd_parts()
...
+    for(i = 1, sl = config.defaultrepo; sl && strcmp(sl->key, config.url.install->str); sl = sl->next) i++;
+    if(sl) default_item = i;
+  }
+
+  i = dia_list(msg, 64, NULL, item_list, default_item, align_left);
+
+  if(i > 0) {
+    if(i == items) {
+      if(!*repo) str_copy(repo, item_list[default_item - 1]);
+      if(dia_input2("Enter the repository URL.", repo, 64, 0)) {
```
Comment (jreidinger) [inline linuxrc.c]: is 64 hard limit or soft width of text entry? As I worry that there can be longer URL.

Comment (wfeldt) [inline linuxrc.c]: it's just the displayed field size

File: linuxrc.c (hunk 2 above)
Comment (wfeldt) [inline linuxrc.c]: true enough; I'll check the workflow tomorrow, I think it's wrong this way

File: linuxrc.c (hunk 4 above)
Comment (jreidinger) [inline linuxrc.c]: then it is fine, I just want to be sure.

**Comment by jreidinger** [comment]
LGTM

**Pull request merged**